From 839b25d6bcef4b1609aaa53d018cdbc59ea4efa7 Mon Sep 17 00:00:00 2001
From: mokopatches <mokopatches@openmoko.org>
Date: Wed, 16 Jul 2008 14:44:50 +0100
Subject: [PATCH] i2c-permit_invalid_addrs.patch
 We need this stupid workaround since our amplifier chip uses a 'reserved' I2C
 address

Signed-off-by: Harald Welte <laforge@openmoko.org>
---
 drivers/i2c/i2c-core.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/i2c-core.c b/drivers/i2c/i2c-core.c
index d0175f4..bb0c5e5 100644
--- a/drivers/i2c/i2c-core.c
+++ b/drivers/i2c/i2c-core.c
@@ -1026,11 +1026,11 @@ static int i2c_probe_address(struct i2c_adapter *adapter, int addr, int kind,
 	int err;
 
 	/* Make sure the address is valid */
-	if (addr < 0x03 || addr > 0x77) {
+	/*if (addr < 0x03 || addr > 0x77) {
 		dev_warn(&adapter->dev, "Invalid probe address 0x%02x\n",
 			 addr);
 		return -EINVAL;
-	}
+	}*/
 
 	/* Skip if already in use */
 	if (i2c_check_addr(adapter, addr))
-- 
1.5.6.3

